src/utils/MainApi.js

```diff
@@ -0,0 +1,65 @@
+import { checkResponseStatus } from "./utils";
+
+const baseURL = "https://api.dean-news.students.nomoredomainssbs.ru";
+
+const MainApi = {
+  register: (data) => {
+    return fetch(`${baseURL}/signup`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(data),
+    }).then(checkResponseStatus);
```

Comment:
**Might be better**

You can make a special method for fetching and checking responses **not to duplicate** it in every request:

```js

  function request(url, options) {
    return fetch(url, options).then(checkResponseStatus)
  }

```

And now you can safely replace `fetch` with `request`. And you don’t need to change anything else.